subxt/src/client/light_client/builder.rs

```diff
@@ -88,7 +89,14 @@ impl<T: Config> LightClientBuilder<T> {
     ///
     /// ## Panics
     ///
-    /// Panics if being called outside of `tokio` runtime context.
+    /// ### Native
+    ///
+    /// Panics when called outside of `tokio` runtime context for native context.
+    ///
+    /// ### Web
+    ///
+    /// If smoldot panics, then the promise created will be leaked. For more details, see
+    /// https://docs.rs/wasm-bindgen-futures/latest/wasm_bindgen_futures/fn.future_to_promise.html.
```

Comment:
```suggestion
    /// The panic behaviour depends on the feature flag being used:
    ///
    /// ### Native
    ///
    /// Panics when called outside of a `tokio` runtime context.
    ///
    /// ### Web
    ///
    /// If smoldot panics, then the promise created will be leaked. For more details, see
    /// https://docs.rs/wasm-bindgen-futures/latest/wasm_bindgen_futures/fn.future_to_promise.html.
```

(Am I right in understanding that for the `web` feature flag, it won't panic at all (under any normal operaiton), and if smoldot panics then it will just never return anything (hence the promise might live forever?)

Reply:
Yep, that is my understanding too. Our wasm module will "panic", which is translated into a javascript exception; and everything will continue to "work" as normal, although we'll not have a smoldot on a background task anymore 


> Note that in wasm panics are currently translated to aborts, but “abort” in this case means that a JavaScript exception is thrown. The wasm module is still usable (likely erroneously) after Rust panics.
> If the future provided panics then the returned Promise will not resolve. Instead it will be a leaked promise. This is an unfortunate limitation of wasm currently that’s hoped to be fixed one day!